ambry-tools/src/main/java/com/github/ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -60,11 +60,18 @@ public static void main(String[] args) throws IOException {
     OptionSet options = parser.parse(args);
 
     String[] destZkAndCluster = options.valueOf(destOpt).split(HelixVcrUtil.SEPARATOR);
-    if (destZkAndCluster.length != 3) {
-      errorAndExit("dest argument must have form 'zkString/namespace/clusterName'");
+    String destZkString;
+    String destClusterName;
+    if (destZkAndCluster[0].contains("shared")) {
+      if (destZkAndCluster.length != 3) {
+        errorAndExit("dest argument must have form 'zkString/namespace/clusterName'");
+      }
+      destZkString = destZkAndCluster[0] + HelixVcrUtil.SEPARATOR + destZkAndCluster[1];
+      destClusterName = destZkAndCluster[2];
```

Comment:
I don't think we need namespace in the src and dest zookeeper address. We would just take whatever address provided in the argument. If ambry's zookeeper znode exists under a chroot, we should make sure it's provided in the argument.

Reply:
ack

Reply:
chroot == namespace.

On old zk clusters, the znode is like zk_string/cluster_name
But on new (shared) zk clusters, the znode is like zk_string/ambry/clustername

clustername is always the last arg.
but the prefix to that varies.

Reply:
this came about when working with shared ZK clusters

Reply:
done, changed the code to excl namesapce